Realm/RLMResults.mm

```diff
@@ -312,6 +314,109 @@ - (RLMResults *)sortedResultsUsingDescriptors:(NSArray *)properties {
     return [RLMResults resultsWithObjectClassName:self.objectClassName query:move(query) sort:sorter realm:_realm];
 }
 
+namespace {
+    struct RealmCreation {
+        NSString *path;
+        NSData *key;
+        BOOL readOnly;
+        BOOL inMemory;
+        BOOL dynamic;
+        RLMSchema *schema;
+    };
+
+    RLMRealm *createWithRealmCreation(RealmCreation realmCreation, SharedGroup::VersionID version) {
+        try {
+            NSError *realmCreationError = nil;
+            RLMRealm *realm = [RLMRealm realmWithPath:realmCreation.path key:realmCreation.key readOnly:realmCreation.readOnly inMemory:realmCreation.inMemory dynamic:realmCreation.dynamic schema:realmCreation.schema error:&realmCreationError];
```

Comment:
we probably want to pass `nil` for the error pointer here so that we throw if the realm can't be created.